tests/test_odp_manager.py

```diff
@@ -0,0 +1,386 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from unittest import mock
+
+from optimizely import exceptions as optimizely_exception
+from optimizely import version
+from optimizely.helpers.enums import Errors
+from optimizely.odp.lru_cache import OptimizelySegmentsCache, LRUCache
+from optimizely.odp.odp_config import OdpConfig
+from optimizely.odp.odp_event_manager import OdpEventManager
+from optimizely.odp.odp_manager import OdpManager
+from optimizely.odp.odp_segment_manager import OdpSegmentManager
+from optimizely.odp.zaius_graphql_api_manager import ZaiusGraphQLApiManager
+from optimizely.odp.zaius_rest_api_manager import ZaiusRestApiManager
+from tests import base
+
+
+class OdpManagerTest(base.BaseTest):
+
+    def test_configurations_disable_odp(self):
+        mock_logger = mock.MagicMock()
+        manager = OdpManager(True, OptimizelySegmentsCache, None, None, mock_logger)
+
+        mock_logger.info.assert_called_once_with('ODP is disabled.')
+        manager.update_odp_config('valid', 'host', [])
+        self.assertIsNone(manager.odp_config.get_api_key())
+        self.assertIsNone(manager.odp_config.get_api_host())
+
+        manager.fetch_qualified_segments('user1', [])
+        mock_logger.error.assert_called_once_with(Errors.ODP_NOT_ENABLED)
+
+        # these call should be dropped gracefully with None
+        manager.identify_user('user1')
+
+        self.assertRaisesRegex(optimizely_exception.OdpNotEnabled, Errors.ODP_NOT_ENABLED,
+                               manager.send_event, 't1', 'a1', {}, {})
+
+        self.assertIsNone(manager.event_manager)
+        self.assertIsNone(manager.segment_manager)
+
+    def test_fetch_qualified_segments(self):
+        mock_logger = mock.MagicMock()
+        segment_manager = OdpSegmentManager(OdpConfig(), OptimizelySegmentsCache,
+                                            ZaiusGraphQLApiManager(mock_logger), mock_logger)
+
+        manager = OdpManager(False, OptimizelySegmentsCache, segment_manager, None, mock_logger)
+
+        with mock.patch.object(segment_manager, 'fetch_qualified_segments') as mock_fetch_qualif_segments:
+            manager.fetch_qualified_segments('user1', [])
+
+        mock_logger.debug.assert_not_called()
+        mock_logger.error.assert_not_called()
+        mock_fetch_qualif_segments.assert_called_once_with('fs_user_id', 'user1', [])
+
+        with mock.patch.object(segment_manager, 'fetch_qualified_segments') as mock_fetch_qualif_segments:
+            manager.fetch_qualified_segments('user1', ['IGNORE_CACHE'])
+
+        mock_logger.debug.assert_not_called()
+        mock_logger.error.assert_not_called()
+        mock_fetch_qualif_segments.assert_called_once_with('fs_user_id', 'user1', ['IGNORE_CACHE'])
+
+    def test_fetch_qualified_segments__seg_mgr_and_seg_cache_are_none(self):
```

Comment:
@andrewleap-optimizely 
Can you check this test. If I assign None to manager.OptimizelySegmentsCache and manager.segment_manager I get ODP is not enabled. Which I think is a good test. 
But I would think users would pass cache and seg mgr as None in OdpManager constructor -> that gives a different error, which is api_key and host_key are None (from OdpConfig), that's because OdpConfig is dummy, without values.
Which test is more accurate/ correct? The goal is to test a scenario where self.segment_manager and segments_cache are None (line 52, 53 in the constructor)
Similarly in the test "test_fetch_qualified_segments__not_enabled" I make manager.enabled = False. But maybe I would need to set Odpmanager(disabled=True...)?

Alternatively I can remove this test.

Reply:
This is definitely a good test! You're on the right track by omitting segments_cache and segment_manager. The part you're missing is simulating odp_config being retrieved before running fetch_qualified_segments.
```python
manager = OdpManager(False, logger=mock_logger)
manager.update_odp_config('api_key', 'api_host', [])
```

Reply:
As for `test_fetch_qualified_segments__not_enabled`, you should definitely test with `disabled=True`. However to be a valid assertion, you need to move the call to `fetch_qualified_segments` into the patch context block.